src/univariate/discrete/discreteuniform.jl

```diff
@@ -21,21 +21,35 @@ External links
 
 * [Discrete uniform distribution on Wikipedia](http://en.wikipedia.org/wiki/Uniform_distribution_(discrete))
 """
-struct DiscreteUniform <: DiscreteUnivariateDistribution
-    a::Int
-    b::Int
+struct DiscreteUniform{I} <: DiscreteUnivariateDistribution where I
+    a::I
+    b::I
     pv::Float64 # individual probabilities
 
-    function DiscreteUniform(a::Real, b::Real; check_args::Bool=true)
+    function DiscreteUniform{I}(a::I, b::I; check_args::Bool=true) where I <: Integer
         @check_args DiscreteUniform (a <= b)
-        new(a, b, 1 / (b - a + 1))
+        new{I}(a, b, one(I) / (b - a + one(I)))
+    end
+    function DiscreteUniform(a::Real, b::Real; kwargs...)
+        a_int = a isa Integer ? a : ceil(Int,a)
+        b_int = b isa Integer ? b : floor(Int, b)
+        aI, bI = promote(a_int,b_int)
+        I = typeof(aI)
+        DiscreteUniform{I}(aI,bI; kwargs...)
     end
     DiscreteUniform(b::Real; check_args::Bool=true) = DiscreteUniform(0, b; check_args=check_args)
-    DiscreteUniform() = new(0, 1, 0.5)
+    DiscreteUniform() = new{Int}(0, 1, 0.5)
 end
```

Comment:
Can we avoid such major changes in the PR?

Reply:
Originally, I introduced those modifications to discreteuniform to test how in addition to Float64/Float32 also Integer parameters could introduce a type parameter to allow using Float64/Float32. Since, this blows up the pull-request, I will move them to its own PR after this PR is hopefully accepted.